commands/verify/verify.go

```diff
@@ -84,165 +106,243 @@ func verifySingleCase(expectedFile, actualFile, query string) error {
+			continue
+		}
+
+		for current := 0; current <= verifyInfo.retryCount; current++ {
+			if err := verifySingleCase(v.GetExpected(), v.GetActual(), v.Query); err == nil {
+				summary.successNum++
+				if current == 0 {
+					spinnerLiveText.Success(fmt.Sprintf("verified %v \n", caseName(v)))
+				} else {
+					spinnerLiveText.Success(fmt.Sprintf("verified %v, retried %d time(s)\n", caseName(v), current))
+				}
+				break
+			} else if current != verifyInfo.retryCount {
+				if current == 0 {
+					spinnerLiveText.UpdateText(fmt.Sprintf("failed to verify %v, will continue retry:", caseName(v)))
+					time.Sleep(time.Second * 2)
+				} else {
+					spinnerLiveText.UpdateText(fmt.Sprintf("failed to verify %v, retry [%d/%d]", caseName(v), current, verifyInfo.retryCount))
+					time.Sleep(verifyInfo.interval)
+				}
+			} else {
+				summary.errNum++
+				spinnerLiveText.UpdateText(fmt.Sprintf("failed to verify %v, retry [%d/%d]", caseName(v), current, verifyInfo.retryCount))
+				time.Sleep(time.Second)
+				spinnerLiveText.Warning(fmt.Sprintf("failed to verify %v, retried %d time(s):", caseName(v), current))
+				spinnerLiveText.Fail(err)
+				fmt.Println()
+				if verifyInfo.failFast {
+					outputSummary(&summary, verifyInfo.caseNumber)
+					return fmt.Errorf("failed to verify one case, an error occurred")
+				}
 			}
 		}
-		wg.Done()
-	}()
+	}
 
-	if v.GetExpected() == "" {
-		errMsg := fmt.Sprintf("the expected data file for case[%v] is not specified\n", idx)
-		logger.Log.Warnf(errMsg)
-		err = errors.New(errMsg)
-		return
+	outputSummary(&summary, verifyInfo.caseNumber)
+	if summary.errNum > 0 {
+		return fmt.Errorf("failed to verify %d case(s)", summary.errNum)
 	}
+	return nil
+}
 
-	for current := 1; current <= verify.retryCount; current++ {
-		if err = verifySingleCase(v.GetExpected(), v.GetActual(), v.Query); err == nil {
-			break
-		} else if current != verify.retryCount {
-			logger.Log.Warnf("verify case[%v] failure, will continue retry, %v", idx, err)
-			time.Sleep(verify.interval)
+// outputSummary outputs a summary of verify result. The summary shows the number of the successful, failed and skipped cases.
+func outputSummary(summary *Summary, total int) {
+	pterm.Info.Prefix = pterm.Prefix{
+		Text:  "SUMMARY",
+		Style: &pterm.ThemeDefault.InfoPrefixStyle,
+	}
+	pterm.Info.WithMessageStyle(&pterm.Style{pterm.FgGreen}).Println(fmt.Sprintf("%d passed", summary.successNum))
+	pterm.Info.Prefix = pterm.Prefix{
+		Text:  "       ",
+		Style: &pterm.ThemeDefault.InfoPrefixStyle,
+	}
+	pterm.Info.WithMessageStyle(&pterm.Style{pterm.FgLightRed}).Println(fmt.Sprintf("%d failed", summary.errNum))
+	pterm.Info.WithMessageStyle(&pterm.Style{pterm.FgYellow}).Println(fmt.Sprintf("%d skipped", total-summary.errNum-summary.successNum))
+	fmt.Println()
+}
+
+// outputResult outputs the result of cases.
+func outputResult(outputInfo *OutputInfo, summary *Summary) {
+	spinnerLiveText, _ := pterm.DefaultSpinner.WithShowTimer(false).Start()
```

Comment:
```suggestion
```

Reply:
This line can be removed, since we only call this function in concurrent mode, and results will be printed at once.